skyportal/handlers/api/classification.py

```diff
@@ -1,37 +1,138 @@
+              Arrow-parseable date string (e.g. 2020-01-01). If provided, filter by
+              last_detected_at >= startDate
+          - in: query
+            name: endDate
+            nullable: true
+            schema:
+              type: string
+            description: |
+              Arrow-parseable date string (e.g. 2020-01-01). If provided, filter by
+              last_detected_at <= endDate
+          - in: query
+            name: numPerPage
+            nullable: true
             schema:
               type: integer
-        responses:
-          200:
-            content:
-              application/json:
-                schema: SingleClassification
-          400:
-            content:
-              application/json:
-                schema: Error
+            description: |
+              Number of sources to return per paginated request. Defaults to 100. Max 500.
+          - in: query
+            name: pageNumber
+            nullable: true
+            schema:
+              type: integer
+            description: Page number for paginated query results. Defaults to 1
+          responses:
+            200:
+              content:
+                application/json:
+                  schema:
+                    allOf:
+                      - $ref: '#/components/schemas/Success'
+                      - type: object
+                        properties:
+                          data:
+                            type: object
+                            properties:
+                              sources:
+                                type: array
+                                items:
+                                  $ref: '#/components/schemas/Classification'
+                              totalMatches:
+                                type: integer
+            400:
+              content:
+                application/json:
+                  schema: Error
+
         """
-        classification = Classification.get_if_accessible_by(
-            classification_id, self.current_user, raise_if_none=True
+
+        page_number = self.get_query_argument('pageNumber', 1)
+        n_per_page = min(
+            int(
+                self.get_query_argument("numPerPage", DEFAULT_CLASSIFICATIONS_PER_PAGE)
+            ),
+            MAX_CLASSIFICATIONS_PER_PAGE,
         )
-        return self.success(data=classification)
+        start_date = self.get_query_argument('startDate', None)
+        end_date = self.get_query_argument('endDate', None)
+
+        if classification_id is not None:
+            classification = Classification.get_if_accessible_by(
+                classification_id, self.current_user, raise_if_none=True
+            )
+            return self.success(data=classification)
+
+        # get owned
+        classifications = Classification.query_records_accessible_by(self.current_user)
+
+        if start_date:
+            start_date = str(arrow.get(start_date.strip()).datetime)
```

Comment:
Do we want to filter using a datetime object or a string?

Reply:
This is fine. Turns out either works.